src/plugins/telemetry/server/plugin.ts

```diff
@@ -232,7 +238,17 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
       return;
     }
 
-    const config = await firstValueFrom(this.config$);
+    const configMaybe = await withTimeout<TelemetryConfigType>({
+      promise: firstValueFrom(this.config$),
+      timeoutMs: 3000,
+    });
```

Comment:
Why 3s? AFAIK, `this.config$` should be immediate (we use the subscriber to account for config reloads)

Reply:
If the config does not change no new emissions will happen. Thus, the RxJS pipeline defined in the constructor will get stuck, cause we are using `exhaustMap`.

I was using this 3 second timeout to unblock the pipeline and proceed to the next emission of the `timer()`, which will `takeUntil(this.pluginStop$)` and complete the RxJS pipeline.

Perhaps another option would be to use `switchMap` instead of `exhaustMap`, this way, we would be cancelling the `firstValueFrom(this.config$)` every `OPT_IN_POLL_INTERVAL_MS` if it hasn't finished. WDYT?

Reply:
Ideally, the `config$` Observable that we obtain from `ConfigService` should complete on stop, but this does not seem to be the case ATM.

Reply:
FYI I just tried the `switchMap` approach and it works. It is a bit simpler and does not require this 3 seconds timeout, so I updated the PR with these changes.